File: Which_Fuel_is_Cheaper.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{

    int t;
    cin >> t;

    while (t--)
    {
        int x, y, a, b, k;
        cin >> x >> y >> a >> b >> k;
        long long petrol = x;
        long long diesel = y;
        for (int i = 0; i < k; i++)
        {
            petrol += a;
            diesel += b;
        }

        if (petrol == diesel)
        {
            cout << "SAME PRICE"<< endl;
        }
        else if (petrol > diesel)
        {
            cout << "DIESEL" << endl;
        }
        else
        {
            cout << "PETROL" << endl;
        }
    }
    return 0;
}

File: Which_Fuel_is_Cheaper.java
import java.util.Scanner;

public class CodeChef{
    public static void main(String[] args) {
        int t;
        Scanner sc = new Scanner(System.in);
        t = sc.nextInt();

        while(t-- != 0){
            int x, y, a, b, k;

            x = sc.nextInt();
            y = sc.nextInt();
            a = sc.nextInt();
            b = sc.nextInt();
            k = sc.nextInt();
            long petrol = x + a*k;
            long diesel = y + b*k; 
            

            if(petrol == diesel){
                System.out.println("SAME PRICE");
            }else if( petrol > diesel){
                System.out.println("DIESEL");
            }else{
                System.out.println("PETROL");
            }
        }
        sc.close();
    }
}